### PR TITLE
Add joint genotyping

### PR DESCRIPTION
**Pull request by vladsavelyev**
Add joint genotyping jobs:
* Preparing intervals for running jobs in parallel,
* Loading GVCFs into [GenomicsDB](https://gatk.broadinstitute.org/hc/en-us/articles/360035891051-GenomicsDB) split by intervals
* Running [GATK GenotypeGVCFs](https://gatk.broadinstitute.org/hc/en-us/articles/360037057852-GenotypeGVCFs) on DBs,
* Merging into a joint-called VCF,
* Making site-only joint-called VCF in preparation for VQSR.

### CODE REVIEW

File: jobs/joint_genotyping.py [hunk 1]
```diff
@@ -0,0 +1,735 @@
...
+from cpg_utils.workflows.filetypes import GvcfPath
+
+from .vcf import gather_vcfs
+from .picard import get_intervals
+
+
+class JointGenotyperTool(Enum):
+    """
+    Tool used for joint genotyping. GenotypeGVCFs is more stable,
+    GnarlyGenotyper is fater but more experimental.
```
Comment (lgruen) [inline jobs/joint_genotyping.py:30]:
```suggestion
    GnarlyGenotyper is faster but more experimental.
```

File: jobs/joint_genotyping.py [hunk 2]
```diff
@@ -0,0 +1,735 @@
...
+        '--consolidate',
+        # The batch_size value was carefully chosen here as it is the optimal value for
+        # the amount of memory allocated within the task; please do not change it
+        # without consulting the Hellbender (GATK engine) team!
+        '--batch-size',
+        '50',
+        # Using cloud + --consolidate causes a TileDB error:
+        # https://github.com/broadinstitute/gatk/issues/7653
+        # However disabling --consolidate decreases performance of reading, cause
+        # GenotypeGVCFs to run forever. So instead, using non-cloud GenomicsDB.
```
Comment (lgruen) [inline jobs/joint_genotyping.py:509]: Did you mean to put this comment above `--consolidate` or maybe have a commented-out `--cloud` option?

You also mention this already in line 100, should that comment be removed?

File: jobs/joint_genotyping.py [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+
+    GnarlyGenotyper is an experimental GATK joint-genotyping tool that performs
+    "quick and dirty" joint genotyping on large cohorts, of GVCFs post-processed
+    with ReblockGVCF.
+
+    HaplotypeCaller must be used with `-ERC GVCF` or `-ERC BP_RESOLUTION` to add
+    genotype likelihoods.
+
+    ReblockGVCF must be run to add all the annotations necessary for VQSR:
+    QUALapprox, VarDP, RAW_MQandDP.
```
Comment (lgruen) [inline jobs/joint_genotyping.py:560]: I'm still puzzled why that's part of `ReblockGVCF`... ;)

**Review by lgruen — APPROVED** [review]

File: jobs/joint_genotyping.py (hunk 1 above)
Comment (vladsavelyev) [reply jobs/joint_genotyping.py:30]: Fatter 😬

File: jobs/joint_genotyping.py (hunk 2 above)
Comment (vladsavelyev) [reply jobs/joint_genotyping.py]: I can actually just remove all the code for interacting with the GenomicsDB on cloud, we are not using it anyway. I'll move this comment into the docstring so we don't forget why we are not using the cloud mode.

File: jobs/joint_genotyping.py (hunk 3 above)
Comment (vladsavelyev) [reply jobs/joint_genotyping.py:326]: GATK acts in mysterious ways.

Comment (vladsavelyev) [reply jobs/joint_genotyping.py:326]: It would be amazing though in some future to just switch to Hail Combiner for this

**Pull request merged**